src/hooks/useFetchRecommendationPod.ts

```diff
@@ -54,7 +54,7 @@ const useFetchRecommendationPod = (
     if (fetchZeroStateOnFocus) return;
     fetchRecommendationResults();
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [cioClient]);
+  }, [cioClient, recommendationPods]);
```

Comment:
This is the main fix right? I got confused because of the other changes, wasn't sure how they were fixing the issue and missed this.

Correct me if I am wrong, but the change sin useSections don't change any behavior.

Reply:
You are correct I'm removing the unnecessary code changes